gravityflow/gravityflow
# HS6254 - Merge tag for {current_step} with modifiers

This is a minor enhancement which creates a new merge tag to allow various data about the current step to be displayed via merge tag.

**New Tag + Modifiers**
* {current_step}                       //Defaults to :name
* {current_step:ID}
* {current_step:name}
* {current_step:type}
* {current_step:schedule}       //Date/Time step is scheduled to start
* {current_step:start}              //Date/Time step execution was started
* {current_step:expiration}     //Date/Time step is scheduled to expire
* {current_step:duration}        //Formatted time length from start of step to now
* {current_step:duration_seconds}        //Non-Formatted time length from start of step to now in seconds
* {current_step:duration_minutes}        //Non-Formatted time length from start of step to now in mins

Any modifier data value which does not exist (step not scheduled or has no expiration) will just remove the merge tag from the parsed response. i.e. does not put 'No expiry defined' or the like.

**To Test**
You can use any of the merge tag details described above in any location where merge tags will process. [This form export](https://github.com/gravityflow/gravityflow/files/2195341/gravityforms-export-2018-07-14.json.zip) includes 6 steps covering approval and user input step types with and without schedule, expiration.

If you want to see all variations via the inbox / status list table, the following snippet will replace the submitter name column and process merge tags on the filter response before returning to screen.
```<?php
add_filter( 'gravityflow_status_submitter_name', 'inbox_submitter_column_info', 10, 3 );
add_filter( 'gravityflow_inbox_submitter_name', 'inbox_submitter_column_info', 10, 3 );
function inbox_submitter_column_info( $column_value, $entry, $form ) {
	$text = 'Default (Name): {current_step}<br/>';
	$text .= 'ID: {current_step:ID}<br/>';
	$text .= 'Name: {current_step:name}<br/>';
	$text .= 'Type: {current_step:type}<br/>';
	$text .= 'Schedule: {current_step:schedule}<br/>';
	$text .= 'Start: {current_step:start}<br/>';
	$text .= 'Expiration: {current_step:expiration}<br/>';
	$text .= 'Duration: {current_step:duration}<br/>';
	$text .= 'Duration_s: {current_step:duration_seconds}<br/>';
	$text .= 'Duration_m: {current_step:duration_minutes}<br/>';
	$output = GFCommon::replace_variables( $text, $form, $entry, false, true );

	$column_value = $output;
	return $column_value;
}
?>
```
